File: SIMINOV-CONNECT-TEMPLATE/src/siminov/connect/template/services/AddLiquor.java
/** 
 * [SIMINOV FRAMEWORK]
 * Copyright [2015] [Siminov Software Solution LLP|support@siminov.com]
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 **/

package siminov.connect.template.services;

import siminov.connect.connection.design.IConnectionRequest;
import siminov.connect.connection.design.IConnectionResponse;
import siminov.connect.exception.ServiceException;
import siminov.connect.service.Service;
import siminov.connect.template.model.Liquor;
import siminov.connect.template.writter.LiquorWritter;
import siminov.orm.exception.DatabaseException;
import siminov.orm.log.Log;

public class AddLiquor extends Service {

	private final String SERVICE_NAME = "SIMINOV-CONNECT-LIQUORS-SERVICE";
	private final String API_NAME = "ADD-LIQUOR";

	public static final String LIQUOR = "LIQUOR";
	
	public AddLiquor() {
		setService(SERVICE_NAME);
		setApi(API_NAME);
	}
	
	public void onStart() {
		
	}

	public void onQueue() {
		
	}

	public void onPause() {
		
	}

	public void onResume() {
		
	}

	public void onFinish() {
		
	}

	public void onApiInvoke(IConnectionRequest connectionRequest) {
		
		if(connectionRequest.getDataStream() == null) {
			return;
		}
		
		
		String liquorType = (String) getResource(LIQUOR);
		Liquor liquor = null;
		
		try {
			Liquor[] liquors = new Liquor().select().where(Liquor.LIQUOR_TYPE).equalTo(liquorType).execute();
			if(liquors != null && liquors.length > 0) {
				liquor = liquors[0];
			}
		} catch(DatabaseException de) {
			Log.error(AddLiquor.class.getName(), "onServiceApiInvoke", "Database Exception caught while getting liquor from database, " + de.getMessage());
			return;
		}
		
		
		LiquorWritter liquorWritter = new LiquorWritter();
		byte[] dataStream = liquorWritter.build(liquor);
		
		connectionRequest.setDataStream(dataStream);
	}

	public void onApiFinish(IConnectionResponse connectionResponse) {
		
	}

	public void onTerminate(ServiceException serviceException) {
		
	}

	public void onRestart() {
		
	}
}

File: Connect-Xamarin-Sample/Droid/Siminov/Connect/Sample/Services/AddLiquor.cs
/** 
 * [SIMINOV FRAMEWORK]
 * Copyright [2015] [Siminov Software Solution LLP|support@siminov.com]
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 **/



using Siminov.Connect.Connection.Design;
using Siminov.Connect.Exception;
using Siminov.Connect.Sample.Model;
using Siminov.Connect.Sample.Writter;
using Siminov.Core.Exception;
using Siminov.Core.Log;
using System;
using System.Collections.Generic;
using System.Text;

namespace Siminov.Connect.Sample.Services
{
	public class AddLiquor : Service.Service
	{
		private readonly String SERVICE_NAME = "SIMINOV-CONNECT-LIQUORS-SERVICE";
		private readonly String REQUEST_NAME = "ADD-LIQUOR";

		public static readonly String LIQUOR = "LIQUOR";

		public AddLiquor() 
		{
			SetService(SERVICE_NAME);
			SetRequest(REQUEST_NAME);
		}

		public override void OnStart() 
		{

		}

		public override void OnQueue() 
		{

		}

		public override void OnPause() 
		{

		}

		public override void OnResume() 
		{

		}

		public override void OnFinish() 
		{

		}

		public override void OnRequestInvoke(IConnectionRequest connectionRequest) 
		{

			if(connectionRequest.GetDataStream() == null) 
			{
				return;
			}


			String liquorType = (String) GetResource(LIQUOR);
			Liquor liquor = null;

			try 
			{
				Liquor[] liquors = (Liquor[]) new Liquor().Select().Where(Liquor.LIQUOR_TYPE).EqualTo(liquorType).Execute();
				if(liquors != null && liquors.Length > 0) 
				{
					liquor = liquors[0];
				}
			} 
			catch(DatabaseException de) 
			{
				Log.Error(typeof(AddLiquor).Name, "onServiceApiInvoke", "Database Exception caught while getting liquor from database, " + de.GetMessage());
				return;
			}


			LiquorWritter liquorWritter = new LiquorWritter();
			byte[] dataStream = liquorWritter.Build(liquor);

			connectionRequest.SetDataStream(dataStream);
		}

		public override void OnRequestFinish(IConnectionResponse connectionResponse) 
		{

		}

		public override void OnTerminate(ServiceException serviceException) 
		{

		}

	}
}
